partest/src/main/scala/scala/tools/nsc/MainGenericRunner.scala

```diff
@@ -83,9 +83,9 @@ class MainGenericRunner {
     val linker = StandardLinker(linkerConfig)
 
     val sjsCode = {
-      val output = WritableMemVirtualJSFile("partest.js")
-      linker.link(ir, moduleInitializers, output, logger)
-      output
+      val out = WritableMemVirtualBinaryFile("partest.js")
+      linker.link(ir, moduleInitializers, Output(out), logger)
+      out
```

Comment:
I think half of this change (*returning* a `WritableMemVirtualBinaryFile` to put in `sjsCode`) should go to the preceding commit (see build failure of that other commit).